overlord/configstate/config/transaction.go

```diff
@@ -103,6 +104,47 @@ func (t *Transaction) Changes() []string {
 	return out
 }
 
+// shadowsVirtualConfig checks that the given subkeys/value does not
+// "block" the path to a virtual config with a non-map type. E.g. if
+// "network.netplan" is virtual it must be impossible to set
+// "network=false" or getting the document under "network" would be
+// wrong.
+func shadowsVirtualConfig(instanceName string, subkeys []string, value interface{}) error {
+	// maps never block the path
+	if v := reflect.ValueOf(value); v.Kind() == reflect.Map {
```

Comment:
do we have cases where value could be a *json.RawMessage? or it doesn't happen at this level?

Reply:
This is only called from transaction.Set() which should never pass *json.RawMessage as values but I added some code now to make this more defensive and report an error if it is.